src/logger/logger.hh

```diff
@@ -0,0 +1,30 @@
+#include "zarr.types.h"
```

Comment:
The fact that these types are not defined within the logger target compromises encapsulation. If the logger is defined as a standalone target, it should be self-contained and reusable. Could we create an additional header file, perhaps named `logger_types.h`, with a C interface to address this issue?